core/storage/btree.rs

```diff
@@ -1542,6 +1543,180 @@ impl BTreeCursor {
         }
         cell_idx
     }
+
+    fn delete_record(&mut self) -> Result<CursorResult<()>> {
+        let mut page_rc = self.stack.top();
+        let page = page_rc.borrow_mut();
+        return_if_locked!(page);
+
+        let contents = page.get().contents.as_mut().unwrap();
+        let cell_idx = self.stack.current_index() as usize;
+        println!("cell_idx={}", cell_idx);
+        println!("cell_count={}", contents.cell_count());
+        if cell_idx >= contents.cell_count() {
+            return Err(LimboError::InternalError(
+                "Delete: invalid cell index".to_string(),
+            ));
+        }
+
+        self.drop_cell(contents, cell_idx);
+
+        page.set_dirty();
+        self.pager.add_dirty(page.get().id);
+
+        return_if_io!(self.balance_after_delete());
```

Comment:
I would need more time to reviews this. As far as I know sqlite deletes a cell from a page and the there is a loop that calls balance. `balance` checks wether there is an overflow (what we do now) or underflow. I want to refactor the balance because it should be as close to sqlite as possible so we should stick to that instead of having different types of balancing.